sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOWriteTest.java

```diff
@@ -391,6 +392,93 @@ public void testTimePartitioning(BigQueryIO.Write.Method insertMethod) throws Ex
     assertEquals(timePartitioning, table.getTimePartitioning());
   }
 
+  @Test
+  public void testClusteringStreamingInserts() throws Exception {
```

Comment:
Probably also add a test that confirms that we fail if withClustering() is set without withTimePartitioning().

Reply:
Added

```
  @Test(expected = IllegalArgumentException.class)
  public void testClusteringThrowsWithoutPartitioning() throws Exception {
```